tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,269 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.ImmutableMap;
+import com.ibm.icu.util.Freezable;
+import com.ibm.icu.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.unicode.cldr.icu.LDMLConstants;
+
+public class SupplementalCalendarData implements Iterable<String> {
+    /** an <era> element */
+    public static class EraData implements Comparable<EraData> {
+        static final int INDEX = 4; // index of our element
+
+        /** the <era> xpath */
+        private final String start;
+
+        private final String end;
+        private final String code;
+        private final int type;
+        private final String calendarType; // for comparison
+        private final GregorianCalendar startCalendar;
+        private final GregorianCalendar endCalendar;
+
+        EraData(final XPathValue xpath) {
+            calendarType = xpath.getAttributeValue(2, LDMLConstants.TYPE);
+            type = Integer.parseInt(xpath.getAttributeValue(INDEX, LDMLConstants.TYPE));
+
+            start = xpath.getAttributeValue(INDEX, LDMLConstants.START);
+            startCalendar = forDateString(start);
+            end = xpath.getAttributeValue(INDEX, LDMLConstants.END);
+            endCalendar = forDateString(end);
+            code = xpath.getAttributeValue(INDEX, LDMLConstants.CODE);
+        }
+
+        public int getType() {
+            return type;
+        }
+
+        public String getStart() {
+            return start;
+        }
+
+        public String getEnd() {
+            return end;
+        }
+
+        public GregorianCalendar getStartCalendar() {
+            return startCalendar;
+        }
+
+        public GregorianCalendar getEndCalendar() {
+            return endCalendar;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public String[] getAliases() {
+            final String s = xpath.getAttributeValue(INDEX, LDMLConstants.ALIASES);
```

Comment:
Didn't notice this earlier. That's an awkward way to get an empty string array. Also, we should avoid String::split.

Use a static Splitter with splitToList, and trimming, and change the return value to a List